CMakeLists.txt

```diff
@@ -107,13 +142,20 @@ endif()
 if(_use_onemkl_interfaces)
     set(BUILD_FUNCTIONAL_TESTS False)
     set(BUILD_EXAMPLES False)
+    set(ENABLE_MKLGPU_BACKEND True)
+    set(ENABLE_MKLCPU_BACKEND True)
+
     if(_use_onemkl_interfaces_cuda)
         set(ENABLE_CUBLAS_BACKEND True)
         set(ENABLE_CUSOLVER_BACKEND True)
         set(ENABLE_CUFFT_BACKEND True)
         # set(ENABLE_CURAND_BACKEND True)
-        set(ENABLE_MKLGPU_BACKEND True)
-        set(ENABLE_MKLCPU_BACKEND True)
+    endif()
+    if(_use_onemkl_interfaces_hip)
+        set(ENABLE_ROCBLAS_BACKEND True)
+        set(ENABLE_ROCSOLVER_BACKEND True)
+        set(ENABLE_ROCFFT_BACKEND True)
+        # set(ENABLE_ROCRAND_BACKEND True)
     endif()
 
```

Comment:
add 'else' branch and check if `_use_onemkl_interfaces_cuda || _use_onemkl_interfaces_hip` then throw A FATAL_ERROR